courses/serializers.py

```diff
@@ -152,6 +159,62 @@ def get_topics(self, instance):
             key=lambda topic: topic["name"],
         )
 
+    def get_page(self, instance):
+        return (
+            CoursePageSerializer(
+                instance=CoursePage.objects.filter(course=instance).get()
+            ).data
+            if CoursePage.objects.filter(course=instance).exists()
+            else None
+        )
+
+    def get_page_url(self, instance):
+        page = self.get_page(instance)
+
+        return None if page is None else page["page_url"]
+
+    def get_description(self, instance):
+        page = CoursePage.objects.filter(course=instance).first()
+        return bleach.clean(page.description, tags=[], strip=True) if page else None
```

Comment:
I have just checked and it seems like we have an options `include_page_fields` in `BaseCourseSerializer` which adds page-related data optionally i.e. page_url, feature_image_src, and financial_assistance_form_url. IMO we should move all these extra page-related fields to `CoursePageSerializer`. Then we can use the serializer context in `CourseViewSet` like:
```
    def get_serializer_context(self):
        context = super().get_serializer_context()
        context["include_page_fields"] = True
        return context
```